ddtrace/tracer/spancontext.go

```diff
@@ -223,13 +223,35 @@ func (t *trace) setTag(key, value string) {
 	t.tags[key] = value
 }
 
+// setPropagatingTag sets the key/value pair as a trace propagating tag.
 func (t *trace) setPropagatingTag(key, value string) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	t.setPropagatingTagLocked(key, value)
+}
+
+// setPropagatingTagLocked sets the key/value pair as a trace propagating tag.
+// Not safe for concurrent use, setPropagatingTag should be used instead in that case.
+func (t *trace) setPropagatingTagLocked(key, value string) {
 	if t.propagatingTags == nil {
 		t.propagatingTags = make(map[string]string, 1)
 	}
 	t.propagatingTags[key] = value
 }
 
+// unsetPropagatingTag deletes the key/value pair from the trace's propagated tags.
+func (t *trace) unsetPropagatingTag(key string) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	t.unsetPropagatingTagLocked(key)
+}
+
+// unsetPropagatingTagLocked deletes the key/value pair from the trace's propagated tags.
+// Not safe for concurrent use, unsetPropagatingTag should be used instead in that case.
+func (t *trace) unsetPropagatingTagLocked(key string) {
+	delete(t.propagatingTags, key)
+}
```

Comment:
I don't think we need this yet anywhere (like we do `setPropagatingTagLocked`), so you can embed it into the host method.